src/plugins/kernel_install/utils.sh

```diff
@@ -289,12 +289,28 @@ function list_installed_kernels()
   local prefix="$4"
   local -a available_kernels=()
   local cmd
+  local file_system_type
+  local ret=0
 
   # TODO: Drop me in the future
   migrate_old_kernel_list
 
-  cmd_manager "$flag" "sudo mkdir -p $REMOTE_KW_DEPLOY"
-  cmd_manager "$flag" "sudo touch $INSTALLED_KERNELS_PATH"
+  file_system_type=$(detect_filesystem_type '')
+  is_filesystem_writable "$file_system_type"
+  if [[ "$?" != 0 ]]; then
+    printf '%s\n' 'WARNING: /boot is read-only. Consider run: kw deploy --setup'
+    ret=30 # EROFS
+  fi
```

Comment:
It may work for first-time users or when completing a cycle of A/B partition switch where the user already know they need to run "kw deploy --setup", but for current kw users it has no effect since the previous script is already there.

Anyway, I don't have a better approach to suggest. At least it will not be an issue for future new users.

Reply:
I'm thinking about that for another PR. I'm thinking about generating an MD5 code for the kw scripts in the target machine and comparing it with the local machine. If the MD5 is different, kw should automatically update the remote with the local version of the scripts.